### PR TITLE
104 post creation location search styling

### PR DESCRIPTION
**Pull request by LordofMankid**
# Description

[REVIEW AND MERGE IN #85 BEFORE THIS ONE, I BUILT IT OFF OF 85 TO AVOID MERGE CONFLICTS. ONCE 85 IS MERGED IN THERE WILL BE LESS FILES CHANGED TO REVIEW]
- frontend changes for the search bar 👯 
- won't allow you to proceed to next step without entering a valid location
- if you switch the name to an invalid location, the last valid location will be saved
- changed posts to show building names, then addresses if no names are available
- map/image rendering behavior is more consistent: only renders map if location is selected, and rerenders carousel if search bar is deselected (unless location is selected)

## Screenshots

IMG_2834
IMG_2835
IMG_2833
IMG_2836

## Possible Downsides

couldn't figure out/was too lazy to manually set and track inputText value so if a location has been selected and you navigate away and back, it'll show you the last set location in the placeholder rather than being a value already set in. didn't really confirm behavior with designers so this could change later

found a bug where if you try to take a picture before it gets ready for the first time, it'll freeze. didn't fix it hehe

### CODE REVIEW

**Comment by LordofMankid** [comment]
idk why screenshots are so big

**Comment by johnny-t06** [comment]
> idk why screenshots are so big

yeah super weird when uploading

File: src/components/common/PlacesSearchBar.tsx [hunk 1]
```diff
@@ -1,54 +1,93 @@
...
-      }}
-      styles={styles}
-      textInputProps={{
-        onFocus: () => onSelected?.(),
-      }}
-    />
+    <>
+      <GooglePlacesAutocomplete
+        placeholder={
+          defaultValue ? defaultValue : "Please enter the building's full name"
```
Comment (johnny-t06) [inline src/components/common/PlacesSearchBar.tsx:19]: Double ??

**Review by johnny-t06 — APPROVED** [review]
LGTM, styling looks great and I tested the posting flow. Good to merge!

**Pull request merged**